backend/src/common/util/workflow_test.go

```diff
@@ -180,48 +180,120 @@ func TestWorkflow_OverrideName(t *testing.T) {
 }
 
 func TestWorkflow_OverrideParameters(t *testing.T) {
-	workflow := NewWorkflow(&workflowapi.Workflow{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "WORKFLOW_NAME",
-		},
-		Spec: workflowapi.WorkflowSpec{
-			Arguments: workflowapi.Arguments{
-				Parameters: []workflowapi.Parameter{
-					{Name: "PARAM1", Value: workflowapi.AnyStringPtr("VALUE1")},
-					{Name: "PARAM2", Value: workflowapi.AnyStringPtr("VALUE2")},
-					{Name: "PARAM3", Value: workflowapi.AnyStringPtr("VALUE3")},
-					{Name: "PARAM4", Value: workflowapi.AnyStringPtr("")},
-					{Name: "PARAM5", Value: workflowapi.AnyStringPtr("VALUE5")},
+	var tests = []struct {
+		name      string
+		workflow  *workflowapi.Workflow
+		overrides map[string]string
+		expected  *workflowapi.Workflow
+	}{
+		{
+			name: "override parameters",
+			workflow: &workflowapi.Workflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "WORKFLOW_NAME",
+				},
+				Spec: workflowapi.WorkflowSpec{
+					Arguments: workflowapi.Arguments{
+						Parameters: []workflowapi.Parameter{
+							{Name: "PARAM1", Value: workflowapi.AnyStringPtr("VALUE1")},
+							{Name: "PARAM2", Value: workflowapi.AnyStringPtr("VALUE2")},
+							{Name: "PARAM3", Value: workflowapi.AnyStringPtr("VALUE3")},
+							{Name: "PARAM4", Value: workflowapi.AnyStringPtr("")},
+							{Name: "PARAM5", Value: workflowapi.AnyStringPtr("VALUE5")},
```

Comment:
nit: does it add anything to have to have all these cases or could re remove `PARAM1` and `PARAM2`

Reply:
It's better to have cases that have focused goals. When they fail, it's easier to find out why.